src/Schema/Types/LaravelEnumType.php

```diff
@@ -43,13 +51,45 @@ function (Enum $enum): array {
                         'name' => $enum->key,
                         'value' => $enum,
                         'description' => $enum->description,
+                        'deprecationReason' => $this->deprecationReason($enum),
                     ];
                 },
                 $enumClass::getInstances()
             ),
         ]);
     }
 
+    protected function deprecationReason(Enum $enum): ?string
+    {
+        $constant = $this->reflection->getReflectionConstant($enum->key);
+        assert($constant instanceof \ReflectionClassConstant, 'Enum keys are derived from the constant names');
+
+        $docComment = $constant->getDocComment();
+        if (false === $docComment) {
+            return null;
+        }
+
+        $docComment = substr($docComment, 3); // strip leading /**
+        $docComment = substr($docComment, 0, -2); // strip trailing */
+
+        $lines = explode("\n", $docComment);
+        foreach ($lines as $line) {
+            $deprecatedPosition = strpos($line, '@' . Directive::DEPRECATED_NAME);
+            if (false === $deprecatedPosition) {
+                continue;
+            }
+
+            // @deprecated is 11 chars long, we want everything after it
+            $reason = trim(substr($line, $deprecatedPosition + 11));
```

Comment:
Let the computer count.
```suggestion
            $deprecatedName = '@' . Directive::DEPRECATED_NAME;
            $reason = trim(substr($line, $deprecatedPosition + strlen($deprecatedName)));
```

Reply:
Found something even simpler, thanks